core/src/test/scala/com/microsoft/ml/spark/codegen/DotnetTestGen.scala

```diff
@@ -0,0 +1,124 @@
+      }
+    }
+  }
+
+  def generateDotnetHelperFile(conf: CodegenConfig): Unit = {
+    val dir = new File(conf.dotnetTestDir,  "mmlsparktest")
+    if (!dir.exists()){
+      dir.mkdirs()
+    }
+    writeFile(new File(dir, "SparkFixtureHelper.cs"),
+      s"""
+         |// Copyright (C) Microsoft Corporation. All rights reserved.
+         |// Licensed under the MIT License. See LICENSE in project root for information.
+         |
+         |using MMLSparktest.Utils;
+         |using Xunit;
+         |
+         |namespace MMLSparktest.Helper
+         |{
+         |    [CollectionDefinition("MMLSpark Tests")]
+         |    public class MMLSparkCollection: ICollectionFixture<SparkFixture>
+         |    {
+         |        // This class has no code, and is never created. Its purpose is simply
+         |        // to be the place to apply [CollectionDefinition] and all the
+         |        // ICollectionFixture<> interfaces.
+         |    }
+         |}
+         |""".stripMargin)
+  }
+
+  // noinspection ScalaStyle
+  def generateDotnetTestProjFile(conf: CodegenConfig): Unit = {
+    val dir = new File(conf.dotnetTestDir,  "mmlsparktest")
+    if (!dir.exists()){
+      dir.mkdirs()
+    }
+    val curProject = conf.name match {
+      case "mmlspark-deep-learning" => "deepLearning"
+      case _ => conf.name.split("-".toCharArray).last
+    }
+    val curPath = conf.dotnetSrcDir.getAbsolutePath
+    val corePath = curPath.replace(curProject, "core")
+    val referenceCore = conf.name match {
+      case "mmlspark-opencv" =>
+        s"""<ProjectReference Include="$corePath\\mmlspark\\CoreProjectSetup.csproj" />"""
+      case _ => ""
+    }
+    writeFile(new File(dir, "TestProjectSetup.csproj"),
+      s"""<Project Sdk="Microsoft.NET.Sdk">
+         |
+         |  <PropertyGroup>
+         |    <TargetFramework>net5.0</TargetFramework>
+         |  </PropertyGroup>
+         |
+         |  <ItemGroup>
+         |    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
+         |    <PackageReference Include="MSTest.TestAdapter" Version="2.1.1" />
+         |    <PackageReference Include="MSTest.TestFramework" Version="2.1.1" />
+         |    <PackageReference Include="coverlet.collector" Version="1.3.0" />
+         |    <PackageReference Include="xunit" Version="2.4.1" />
+         |    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+         |      <PrivateAssets>all</PrivateAssets>
+         |      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+         |    </PackageReference>
+         |    <PackageReference Include="Microsoft.Spark" Version="2.0.0" />
+         |    <PackageReference Include="IgnoresAccessChecksToGenerator" Version="0.4.0" PrivateAssets="All" />
+         |  </ItemGroup>
+         |
+         |  <ItemGroup>
+         |    <ProjectReference Include="..\\..\\..\\..\\..\\..\\..\\..\\core\\src\\main\\dotnet\\dotnetBase.csproj" />
```

Comment:
Generate this path? Also this depends on core is that intended?

Reply:
Sure. Yes because I put those base files of dotnet into core.. Should we move it to an independent folder and let each package depends on it?